grails-app/controllers/org/pih/warehouse/product/ProductController.groovy

```diff
@@ -992,10 +992,15 @@ class ProductController {
      */
     def addSynonymToProduct = {
         println "addSynonymToProduct() " + params
-        def product = Product.get(params.id)
-        if (product) {
-            product.addToSynonyms(new Synonym(name: params.synonym, locale: RCU.getLocale(request)))
-            product.save(flush: true, failOnError: true)
+        Product product = null
+        // Getting locale from RCU, because unable to inject localizationService to productService (session scope)
+        Locale locale = RCU.getLocale(request)
```

Comment:
The user should be choosing which locale, so the locale will be passed in as a parameter.

Reply:
damn, I must've removed it somehow - thanks for vigilance

Reply:
it was obviously meant to be, if the user doesn't provide the locale, it will take it from the current locale, so it was meant to be:
```
Locale locale = params.locale ? new Locale(params.locale) : RCU.getLocale(request)
```

Reply:
Awesome. Now that we ask the user for a locale, if the user doesn't provide one let's throw a validation error. My instinct to use the current locale was a bad one.